### PR TITLE
Add features

### PR DESCRIPTION
**Pull request by zhao-kun**
Full features list:
---

- **Non-intrusive Design**: Zero code modification for Java Spring Cloud application migration, only small configuration update needed.
- **Java Register/Discovery**: Compatible with popular Java Spring Cloud ecosystem's Service register/discovery
  - **Multiple tenants(namespace)** Supporting multiple tenants' service registration, isolation between deferent tenants
  - **Share (global) tenant** Support share tenants, all services have visibility to the service registered in the global tenant.
  - **Compatible** 
    - Be compatible with the eureka registry.
    - Be compatible with the consul registry.
    - Be compatible with the nacos registry.
  - **Extensibility** Support registering services with metadata.
- **Resource Management**: Rely on Kubernetes platform for CPU/Memory resources management.
- **Traffic orchestration** 
	- **Rich Routing Rules:** exact path, path prefix, regular expression of the path, method, headers.
  - **Traffic Splitting** Coloring & Scheduling east-west and north-south traffic to configured services.
  - **LoadBalance** Support Round Robin, Weight Round Robin, Hash by Client IP Address, Hash by HTTP Headers.
- **Resilience**: Including Timeout/CircuitBreaker/Retryer/Limiter, completely follow sophisticated resilience design.
	- **Resilience&Fault Tolerance**
		- **Circuit breaker:** Temporarily blocks possible failures.
		- **Rate limiter:** Limits the rate of incoming requests.
		- **Retryer:** Repeats failed executions.
		- **Time limiter:** Limits the duration of execution.
  - **Chaos engineering**
    - **Fault injection** *Working in progress.*
    - **Delay injection** *Working in progress.*
- **Observability**: 
  - **Logs**
    - **Access Logs** Generate HTTP access log for all request per service
    - **Application log** Automatically inject the tracing context into log data.
  - **Tracing**
    - **JDBC** tracing for invocation of the JDBC.
    - **HTTP Request** tracing for HTTP RPC.
    - **Kafka** tracing for messages delivered by the Kafka.
    - **Redis** tracing for redis cache accessing.
    - **RabbitMQ** tracing for messages delivered by the RabbitMQ.
    - **Sampling**
      - Supporting probabilistic sampling
      - Support QPS sampling
  - **Metrics**
    - **HTTP Request** Reporting throughput latency per URLs
    - **JDBC**  Reporting throughput and latency per SQLs
    - **Kafka** Reporting throughput and latency per consumers, producers and topics
    - **Redis** Reporting throughput and latency per methods
    - **RabbitMQ** Reporting throughput and latency per topics
- **Security**
  - **mTLS** *Working in progress.*
  - **mTLS Enforcement** *Working in progress*
  - **External CA certificate** *Working in progress.*
  - **Service-to-Service Authorization Rules** *Working in progress.*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,13 +65,55 @@ Why do we reinvent another wheel?
 ## 4. Features
 
 - **Non-intrusive Design**: Zero code modification for Java Spring Cloud application migration, only small configuration update needed.
-- **Java Register/Discovery**: Compatible with popular Java Spring Cloud ecosystem's Service register/discovery（Eureka/Consul/Nacos).
-- **Traffic Orchestration**: Coloring & Scheduling east-west and north-south traffic to configured services.
+- **Java Register/Discovery**: Compatible with popular Java Spring Cloud ecosystem's Service registry/discovery
+  - **Multiple tenants(namespace)** Supporting multiple tenants' service registration, isolation between deferent tenants
```
Comment (xxx7xxxx) [inline README.md:69]:
```suggestion
  - **Multiple tenants(namespace)** Supporting multiple tenants' service registration, isolate services from different tenants
```

File: README.md [hunk 2]
```diff
@@ -65,13 +65,55 @@ Why do we reinvent another wheel?
...
+		- **Circuit breaker:** Temporarily blocks possible failures.
+		- **Rate limiter:** Limits the rate of incoming requests.
+		- **Retryer:** Repeats failed executions.
+		- **Time limiter:** Limits the duration of execution.
+  - **Chaos engineering**
+    - **Fault injection** *Working in progress.*
+    - **Delay injection** *Working in progress.*
+- **Observability**: 
+  - **Logs**
+    - **Access Logs** Generate HTTP access log for all request per service
```
Comment (xxx7xxxx) [inline README.md:92]: Please make the ending period `.` and `Capital sentence` below consistent.

File: README.md [hunk 3]
```diff
@@ -65,13 +65,55 @@ Why do we reinvent another wheel?
 ## 4. Features
 
 - **Non-intrusive Design**: Zero code modification for Java Spring Cloud application migration, only small configuration update needed.
-- **Java Register/Discovery**: Compatible with popular Java Spring Cloud ecosystem's Service register/discovery（Eureka/Consul/Nacos).
-- **Traffic Orchestration**: Coloring & Scheduling east-west and north-south traffic to configured services.
+- **Java Register/Discovery**: Compatible with popular Java Spring Cloud ecosystem's Service registry/discovery
+  - **Multiple tenants(namespace)** Supporting multiple tenants' service registration, isolation between deferent tenants
+  - **Share (global) tenant** Support share tenants, all services have visibility to the service registered in the global tenant.
+  - **Compatible** 
+    - Be compatible with the eureka registry.
```
Comment (xxx7xxxx) [inline README.md:72]: Better use `Eureka`, `Consul`, `Nacos` in the documentation.

File: README.md [hunk 4]
```diff
@@ -65,13 +65,55 @@ Why do we reinvent another wheel?
...
+    - Be compatible with the consul registry.
+    - Be compatible with the nacos registry.
+  - **Extensibility** Support registering services with metadata.
 - **Resource Management**: Rely on Kubernetes platform for CPU/Memory resources management.
-- **Canary Deployment**: Routing requests based on colored traffic and different versions of the service.
+- **Traffic orchestration** 
+	- **Rich Routing Rules:** exact path, path prefix, regular expression of the path, method, headers.
+  - **Traffic Splitting** Coloring & Scheduling east-west and north-south traffic to configured services.
+  - **LoadBalance** Support Round Robin, Weight Round Robin, Hash by Client IP Address, Hash by HTTP Headers.
```
Comment (xxx7xxxx) [inline README.md:80]:
```suggestion
  - **LoadBalance** Support Round Robin, Weight Round Robin, Random, Hash by Client IP Address, Hash by HTTP Headers.
```

File: README.md [hunk 5]
```diff
@@ -65,13 +65,55 @@ Why do we reinvent another wheel?
...
+  - **Multiple tenants(namespace)** Supporting multiple tenants' service registration, isolation between deferent tenants
+  - **Share (global) tenant** Support share tenants, all services have visibility to the service registered in the global tenant.
+  - **Compatible** 
+    - Be compatible with the eureka registry.
+    - Be compatible with the consul registry.
+    - Be compatible with the nacos registry.
+  - **Extensibility** Support registering services with metadata.
 - **Resource Management**: Rely on Kubernetes platform for CPU/Memory resources management.
-- **Canary Deployment**: Routing requests based on colored traffic and different versions of the service.
+- **Traffic orchestration**
```
Comment (xxx7xxxx) [inline README.md:77]:
```suggestion
- **Traffic Orchestration** 
```

Consistent with the title.

File: README.md [hunk 6]
```diff
@@ -65,13 +65,55 @@ Why do we reinvent another wheel?
...
+  - **Share (global) tenant** Support share tenants, all services have visibility to the service registered in the global tenant.
+  - **Compatible** 
+    - Be compatible with the eureka registry.
+    - Be compatible with the consul registry.
+    - Be compatible with the nacos registry.
+  - **Extensibility** Support registering services with metadata.
 - **Resource Management**: Rely on Kubernetes platform for CPU/Memory resources management.
-- **Canary Deployment**: Routing requests based on colored traffic and different versions of the service.
+- **Traffic orchestration** 
+	- **Rich Routing Rules:** exact path, path prefix, regular expression of the path, method, headers.
```
Comment (xxx7xxxx) [inline README.md:78]:
```suggestion
	- **Rich Routing Rules:** Exact path, path prefix, regular expression of the path, method, headers.
```

File: README.md [hunk 7]
```diff
@@ -65,13 +65,55 @@ Why do we reinvent another wheel?
...
+    - **Redis** Reporting throughput and latency per methods
+    - **RabbitMQ** Reporting throughput and latency per topics
+- **Security**
+  - **mTLS** *Working in progress.*
+  - **mTLS Enforcement** *Working in progress*
+  - **External CA certificate** *Working in progress.*
+  - **Service-to-Service Authorization Rules** *Working in progress.*
+
+> The throughput is represented by m1, m5, m15
+> The latency is represented by P99, P98, P95, P90, P80, P75, P50, etcd...
```
Comment (xxx7xxxx) [inline README.md:116]:
```suggestion
> The latency is represented by P99, P98, P95, P90, P80, P75, P50, etc...

```

**Review by xxx7xxxx — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -65,12 +65,55 @@ Why do we reinvent another wheel?
...
+		- **Circuit breaker:** Temporarily blocks possible failures.
+		- **Rate limiter:** Limits the rate of incoming requests.
+		- **Retryer:** Repeats failed executions.
+		- **Time limiter:** Limits the duration of execution.
+  - **Chaos engineering**
+    - **Fault injection** *Working in progress.*
+    - **Delay injection** *Working in progress.*
+- **Observability**: 
+  - **Logs**
+    - **Access Logs** Generate HTTP access log for all request per service.
```
Comment (haoel) [inline README.md:92]:
```suggestion
    - **Access Logs** Generate HTTP access log for all requests per service.
```

File: README.md [hunk 9]
```diff
@@ -65,12 +65,55 @@ Why do we reinvent another wheel?
...
+    - **Application log** Automatically inject the tracing context into log data.
+  - **Tracing**
+    - **JDBC** Tracing for invocation of the JDBC.
+    - **HTTP Request** Tracing for HTTP RPC.
+    - **Kafka** Tracing for messages delivered by the Kafka.
+    - **Redis** Tracing for redis cache accessing.
```
Comment (haoel) [inline README.md:98]:
```suggestion
    - **Redis** Tracing for Redis cache accessing.
```

File: README.md [hunk 10]
```diff
@@ -65,12 +65,55 @@ Why do we reinvent another wheel?
...
+    - **Fault injection** *Working in progress.*
+    - **Delay injection** *Working in progress.*
+- **Observability**: 
+  - **Logs**
+    - **Access Logs** Generate HTTP access log for all request per service.
+    - **Application log** Automatically inject the tracing context into log data.
+  - **Tracing**
+    - **JDBC** Tracing for invocation of the JDBC.
+    - **HTTP Request** Tracing for HTTP RPC.
+    - **Kafka** Tracing for messages delivered by the Kafka.
```
Comment (haoel) [inline README.md:97]:
```suggestion
    - **Kafka** Tracing for messages delivered by Kafka.
```

File: README.md [hunk 11]
```diff
@@ -65,12 +65,55 @@ Why do we reinvent another wheel?
...
+    - **RabbitMQ** Tracing for messages delivered by the RabbitMQ.
+    - **Sampling**
+      - Support probabilistic sampling.
+      - Support QPS sampling.
+  - **Metrics**
+    - **HTTP Request** Reporting throughput latency per URLs.
+    - **JDBC**  Reporting throughput and latency per SQLs.
+    - **Kafka** Reporting throughput and latency per consumers, producers and topics.
+    - **Redis** Reporting throughput and latency per methods.
+    - **RabbitMQ** Reporting throughput and latency per topics.
```
Comment (haoel) [inline README.md:108]:
```suggestion
    - **HTTP Request** Reporting throughput latency per URL.
    - **JDBC**  Reporting throughput and latency per SQL.
    - **Kafka** Reporting throughput and latency per consumer, producer, and topic.
    - **Redis** Reporting throughput and latency per method.
    - **RabbitMQ** Reporting throughput and latency per topic.
```

**Review by haoel — APPROVED** [review]

**Pull request merged**